File: dev-docs/snippets/blob-buckets/get-metadata.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Read a blob's metadata (size, content type, tags, sha256) without
// downloading the bytes.
export async function getMetadata(
  client: JsBaoClient,
  bucketIdOrKey: string,
  blobId: string,
) {
  // #region example
  const meta = await client.blobBuckets.getMetadata(bucketIdOrKey, blobId);
  // #endregion example
  return meta;
}

File: dev-docs/snippets/blob-buckets/get-metadata.swift
import JsBaoClient

// Read a blob's metadata without downloading the bytes. Swift returns a
// typed `BucketBlobInfo` (read `meta.numBytes`, `meta.sha256`, `meta.tags`, …).
func getMetadata(
  client: JsBaoClient,
  bucketIdOrKey: String,
  blobId: String
) async throws {
  // #region example
  let meta = try await client.blobBuckets.getMetadata(
    bucketIdOrKey: bucketIdOrKey,
    blobId: blobId
  )
  // #endregion example
  _ = meta
}
